lib/fluent/plugin/in_sql.rb

```diff
@@ -108,6 +108,17 @@ def read_attribute_for_serialization(n)
         end
       end
 
+      # Make sure we always have a Fluent::EventTime object regardless of what comes in
+      def normalized_time(tv, now)
+        return Fluent::EventTime.from_time(tv) if tv.is_a?(Time)
+        begin
+          Fluent::EventTime.parse(tv.to_s)
+        rescue
+          $log.warn "Message contains invalid timestamp, using current time instead (#{now.inspect})"
```

Comment:
Using global `$log` is not good here. Pass `log` to `TableElement` via `init` method and use it.